networks/xml_cnn.py

```diff
@@ -0,0 +1,71 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from networks.base import BaseModel
+
+
+def out_size(l_in, kernel_size, channels, padding=0, dilation=1, stride=1):
+    a = l_in + 2 * padding - dilation * (kernel_size - 1) - 1
+    b = int(a / stride)
+    return (b + 1) * channels
+
+
+class XMLCNN(BaseModel):
+    def __init__(self, config, embed_vecs):
+        super(XMLCNN, self).__init__(config, embed_vecs)
+        assert config.fixed_length is True
+
+        self.filter_sizes = config.filter_sizes
+        emb_dim = embed_vecs.shape[1]
+        num_filter_maps = config.num_filter_maps
+        strides = config.strides
+        d_max_pool_p = config.d_max_pool_p
+
+        self.convs = nn.ModuleList()
+        self.poolings = nn.ModuleList()
+
+        for filter_size, p, stride in zip(self.filter_sizes, d_max_pool_p, strides):
+            conv = nn.Conv1d(
+                in_channels=emb_dim,
+                out_channels=num_filter_maps,
+                kernel_size=filter_size,
+                stride=stride)
+
+            # Dynamic Max-Pooling
+            conv_out_size = out_size(config.max_seq_length, filter_size, num_filter_maps, stride=stride)
```

Comment:
we have conv_out_size and
conv_output_size below. Can just use one name (though for different things)
Maybe conv_output_size we have used that in kimcnn?

Reply:
Solved.